Source/mainmenu.cpp

```diff
@@ -10,6 +10,9 @@ char gszHero[16];
 
 int menu_music_track_id = 5;
 
+int menu_repintro_delay = 30; // seconds
+DWORD dwRepIntro = 0;

```

Comment:
These values should be moved to SourceX/DiabloUI/diabloui.cpp

Reply:
I moved it over and update it given the argument in _UiMainMenuDialog_, as galaxyhaxz pointed out. I kept the default (= 30) to avoid having it happen as soon as the menu opens.
Also moved the `mainmenu_restart_repintro` function there.

Reply:
Ok looking forward to seeing the new code 😊